src/main/java/bio/terra/service/filedata/azure/util/BlobContainerClientFactory.java

```diff
@@ -1,90 +1,71 @@
 package bio.terra.service.filedata.azure.util;
 
+import static bio.terra.common.ValidationUtils.requiresNotBlank;
+
 import com.azure.core.credential.AzureSasCredential;
 import com.azure.core.credential.TokenCredential;
 import com.azure.core.http.HttpClient;
-import com.azure.storage.blob.BlobClient;
 import com.azure.storage.blob.BlobContainerClient;
 import com.azure.storage.blob.BlobContainerClientBuilder;
 import com.azure.storage.blob.BlobServiceClient;
 import com.azure.storage.blob.BlobServiceClientBuilder;
 import com.azure.storage.blob.BlobUrlParts;
-import com.azure.storage.blob.models.UserDelegationKey;
-import com.azure.storage.blob.sas.BlobSasPermission;
-import com.azure.storage.blob.sas.BlobServiceSasSignatureValues;
 import com.azure.storage.common.StorageSharedKeyCredential;
-import com.azure.storage.common.sas.SasProtocol;
-import java.time.OffsetDateTime;
-import java.time.ZoneOffset;
+import java.time.Duration;
 import java.util.Locale;
 import java.util.Objects;
-import org.apache.commons.lang3.StringUtils;
 
 /**
  * A class that wraps the SDK container client and facilities the creation of SAS Urls for blobs by
  * abstracting the different strategies that depend on the authentication mechanism.
  */
 public class BlobContainerClientFactory {
```

Comment:
I still don't think this is a factory class, although now it does seem to contain other factories. After a bit more refactoring its role will probably be more clear.

Reply:
Yes, I agree, and I was going to rename it to `BlobClientWrapper`, but I am still not completely convinced of that name either. So I left it as is, also thinking that further refactoring will provide more clarity in the future -any suggestions would be welcome though.

Reply:
That seems reasonable, as long as we rememer to rename. I don't mind the `BlobContainerClientWrapper` name